### PR TITLE
Pagination available flights

### PR DESCRIPTION
**Pull request by IliaStepanov**

### CODE REVIEW

**Comment by IliaStepanov** [comment]
[[BE] paging in Flight tree](https://trello.com/c/JbWgMPE3/88-be-paging-in-flight-tree)

File: src/main/java/com/epam/lowcost/util/EndPoints.java [hunk 1]
```diff
@@ -19,6 +19,7 @@
     public static final String NEW_TICKET = "/new-ticket";
     public static final String RETURN = "/return";
     public static final String FLIGHT = "/flight";
+    public static final String SET_FLIGHT_BY_PAGE = "/setFlightsByPage";
```
Comment (vakhonin) [inline src/main/java/com/epam/lowcost/util/EndPoints.java]: why camelCase ???????
+ u can use only `/page`

Comment (IliaStepanov) [reply src/main/java/com/epam/lowcost/util/EndPoints.java]: Changed all /entity-by-page to simply /page

File: src/main/java/com/epam/lowcost/DAO/implementations/UserDAOImpl.java [hunk 2]
```diff
@@ -74,16 +74,12 @@ public String deleteUser(long userId) throws DatabaseErrorException {
 
 
     @Override
-    public List<User> getUsersByPage(int pageId, int total) {
+    public List<User> getUsersByPage(int pageId, int usersByPage) {
```
Comment (vakhonin) [inline src/main/java/com/epam/lowcost/DAO/implementations/UserDAOImpl.java]: if `usersByPage` is a number of users on the page, you should give corresponding name

File: src/main/java/com/epam/lowcost/controller/FlightController.java [hunk 3]
```diff
@@ -60,21 +74,35 @@ public String addNewFlight() {
         return "addFlight";
     }
 
-    @GetMapping(value = FLIGHT)
-    public String searchForFlight(Model model) {
-        model.addAttribute("flights", ((FlightServiceImpl) flightService).getAllFlightsWithUpdatedPrice());
+    @GetMapping(value = FLIGHT+ "/{pageId}")
+    public String searchForFlight(@PathVariable int pageId, ModelMap model) {
+        //model.addAttribute("flights", ((FlightServiceImpl) flightService).getAllFlightsWithUpdatedPrice());
```
Comment (vakhonin) [inline src/main/java/com/epam/lowcost/controller/FlightController.java]: remove

File: src/main/java/com/epam/lowcost/controller/FlightController.java [hunk 4]
```diff
@@ -60,21 +74,35 @@ public String addNewFlight() {
...
+
+        int flightsByPage = (int) model.getOrDefault("number", 2);
```
Comment (vakhonin) [inline src/main/java/com/epam/lowcost/controller/FlightController.java]: what does it mean, 2 ?

File: src/main/java/com/epam/lowcost/controller/FlightController.java [hunk 5]
```diff
@@ -60,21 +74,35 @@ public String addNewFlight() {
...
+
+        Map<String, Object> pageRepresentation = flightService.getAllFlightsWithUpdatedPrice(pageId, flightsByPage);
+
+        model.addAttribute("pagesNum", pageRepresentation.get("pagesNum"));
+        model.addAttribute("flights", pageRepresentation.get("flights"));
+        model.addAttribute("pageId", pageRepresentation.get("pageId"));
+
+
+        System.out.println("/flight");
```
Comment (vakhonin) [inline src/main/java/com/epam/lowcost/controller/FlightController.java]: blya

Comment (IliaStepanov) [reply src/main/java/com/epam/lowcost/controller/FlightController.java]: Removed it this morning.

File: src/main/java/com/epam/lowcost/controller/FlightController.java (hunk 4 above)
Comment (IliaStepanov) [reply src/main/java/com/epam/lowcost/controller/FlightController.java]: Number of flights to show on one page by default.

File: src/main/java/com/epam/lowcost/controller/FlightController.java (hunk 3 above)
Comment (IliaStepanov) [reply src/main/java/com/epam/lowcost/controller/FlightController.java]: Removed it this morning.

File: src/main/java/com/epam/lowcost/controller/FlightController.java (hunk 4 above)
Comment (vakhonin) [reply src/main/java/com/epam/lowcost/controller/FlightController.java]: move into separate variable, declaration of similar variables should be obvious

File: src/main/java/com/epam/lowcost/DAO/implementations/UserDAOImpl.java (hunk 2 above)
Comment (IliaStepanov) [reply src/main/java/com/epam/lowcost/DAO/implementations/UserDAOImpl.java]: Changed it to `usersOnPage`

File: src/main/java/com/epam/lowcost/controller/FlightController.java (hunk 4 above)
Comment (IliaStepanov) [reply src/main/java/com/epam/lowcost/controller/FlightController.java]: Changed to more appropriative variable names.

File: src/main/java/com/epam/lowcost/util/Constants.java [hunk 6]
```diff
@@ -0,0 +1,12 @@
+package com.epam.lowcost.util;
+
+import lombok.experimental.UtilityClass;
+
+@UtilityClass
+public class Constants {
+    public static int DEFAULT_NUMBER_OF_USERS_ON_PAGE = 5;
```
Comment (vakhonin) [inline src/main/java/com/epam/lowcost/util/Constants.java:7]: why? you can use only one constant for all entities, they all equal 5...

Comment (IliaStepanov) [reply src/main/java/com/epam/lowcost/util/Constants.java:7]: I think its more convenient and readable, as we might want to change default values separately.

**Pull request merged**